### DISCUSSION TITLE
override lwipopts.h

### DISCUSSION THREAD

**Discussion by sporian-smckown** [ssilverman/QNEthernet#21] [Q&A] [answered]
I'm trying to figure out how to have an app-specific lwipopts.h in an Arduino app that uses QNEthernet.

All my work to date is in GCC with a Makefile-based build.  In this case, one simply adds an app-specific include directory ahead of the entry for lwip in the list of includes (`-I` args) supplied to gcc when it compiles sources.

Googling led me to a way to see the GCC commands Arduino invokes (`arduino --verify --verbose <sketch>`) but no suggestion on how to change the includes used.

Any suggestions?

**Comment by ssilverman** [comment]
Change the `lwipopts.h` file in the `libraries/QNEthernet` folder mentioned here: https://github.com/ssilverman/QNEthernet/discussions/17#discussioncomment-3210309

**Reply by sporian-smckown** [reply]
Changing lwipopts.h in the QEthernet library itself  affects the build of every Arduino app using QEthernet compiled thereafter.  The use case I'm trying to address is an application that wants customized memory and feature settings that should not apply globally to other Arduino apps.

**Reply by ssilverman** [reply]
I’m not sure if there’s a per-sketch way to manage libraries. A cursory Google search didn’t turn up anything. I don’t love the global nature of Arduino libraries (and variables). That’s one of the reasons I use PlatformIO (in VSCode). It allows you to use per-project and per-platform-within-a-project libraries. May I suggest switching to PlatformIO instead?

But this is a problem that’s certainly good to think about. If I have time, I’ll investigate if there’s a way your need can be fulfilled within the confines of the Arduino IDE.

**Comment by sporian-smckown** [comment]
Thanks for the suggestion; I'll take a look at PlatformIO.  I'll likely do some more research on the Arduino front as well.  If I find anything interesting I'll let you know.

**Comment by ssilverman** [comment, marked as answer]
See this section of the README for a guide of how to configure macros for the Arduino IDE:
https://github.com/ssilverman/QNEthernet/blob/0825ded00131fb846027641f3fa071ffc26aad0c/README.md#configuring-macros-using-the-arduino-ide

Below that section is a section on how to do it with PlatformIO.

**Reply by ssilverman** [reply]
Note that this points to an older version of the README. I do this with a current snapshot in case internal links change in the future.
